packages/frameworks/src/frameworks.ts

```diff
@@ -262,6 +263,73 @@ export const frameworks = [
       },
     ],
   },
+  {
+    name: 'Astro',
+    slug: 'Astro',
+    demo: 'https://astro-template.vercel.app',
+    logo: 'https://raw.githubusercontent.com/vercel/vercel/main/packages/frameworks/logos/astro.svg',
+    darkModeLogo:
+      'https://raw.githubusercontent.com/vercel/vercel/main/packages/frameworks/logos/astro-dark.svg',
+    tagline:
+      'Astro is a new kind of static site builder for the modern web. Powerful developer experience meets lightweight output.',
+    description: 'An Astro site, using the basics starter kit.',
+    website: 'https://astro.build',
+    envPrefix: 'PUBLIC_',
+    detectors: {
+      every: [
+        {
+          path: 'package.json',
+          matchContent:
+            '"(dev)?(d|D)ependencies":\\s*{[^}]*"astro":\\s*".+?"[^}]*}',
+        },
+      ],
+    },
+    settings: {
+      installCommand: {
+        placeholder: '`yarn install` or `npm install`',
+      },
+      buildCommand: {
+        value: 'astro build',
+        placeholder: '`npm run build` or `astro build`',
+      },
+      devCommand: {
+        value: 'astro dev --port $PORT',
+        placeholder: 'astro dev',
+      },
+      outputDirectory: {
+        value: 'dist',
+      },
+    },
+    dependency: 'astro',
+    getFsOutputDir: async () => 'dist',
+    getOutputDirName: async () => 'dist',
+    defaultRoutes: [
+      {
+        src: '^/dist/(.*)$',
```

Comment:
Is there a double dist directory (nested)? Since the output directory is dist, that will be the root of the deployed website. Routes match at the root of the deployed website (not the repository root)